trin-core/src/portalnet/overlay_service.rs

```diff
@@ -290,9 +294,14 @@ impl OverlayService {
             }
             Ok(None) => {
                 let enrs = self.find_nodes_close_to_content(request.content_key).await;
-                Ok(Content::Enrs(enrs))
+                match enrs {
+                    Ok(val) => Ok(Content::Enrs(val)),
+                    Err(msg) => Err(OverlayRequestError::InvalidRequest(msg.to_string())),
+                }
             }
-            Err(error) => panic!("Unable to respond to FindContent: {}", error),
+            Err(_) => Err(OverlayRequestError::Failure(
+                "Unable to respond to FindContent".to_string(),
+            )),
```

Comment:
I agree that we should not panic here, but I think it would be good to log the error from the `db`.